fe/fe-core/src/main/java/com/starrocks/qe/ShortCircuitExecutor.java

```diff
@@ -86,13 +88,16 @@ public void exec() {
         throw new StarRocksPlannerException("Not implement ShortCircuit Executor class", ErrorType.INTERNAL_ERROR);
     }
 
-    protected static <T> T pick(List<T> collections) {
-        Preconditions.checkArgument(!collections.isEmpty());
-        if (collections.size() == 1) {
-            return collections.get(0);
-        } else {
-            return collections.get(RANDOM.nextInt(collections.size()));
+    protected static Optional<TNetworkAddress> pick(List<com.starrocks.thrift.TNetworkAddress> hosts,
```

Comment:
Just traverse each backend if of this `range` by `range.getLocations().get(i).getBackend_id()`, and return the first backend which is included by `idToBackend` and is available.

And `range.getLocations()` needn't be shuffled randomly, because this already be shuffle in `OlapScanNode::addScanRangeLocations`.